### PR TITLE
linked responds bugfix

### PR DESCRIPTION
**Pull request by Sleepychord**
fix the bug when receiving linked responds

### CODE REVIEW

**Comment by Sleepychord** [comment]
Maybe there is something wrong in Travis CI......? It mistook variables' scope!

File: src/MFRC522Extended.cpp [hunk 1]
```diff
@@ -828,19 +824,19 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
...
 	// Send an ACK to receive more data
 	// TODO: Should be checked I've never needed to send an ACK
+
 	while (in.prologue.pcb & 0x10) {
 		byte ackData[FIFO_SIZE];
 		byte ackDataSize = FIFO_SIZE;
-
-		result = TCL_TransceiveRBlock(tag, true, ackData, &ackDataSize);
+		byte linked = 0;
+		result = TCL_TransceiveRBlock(tag, true, ackData, &ackDataSize, linked);
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: `linked` should be a pointer

File: src/MFRC522Extended.h [hunk 2]
```diff
@@ -93,7 +93,7 @@ class MFRC522Extended : public MFRC522 {
 	/////////////////////////////////////////////////////////////////////////////////////
 	StatusCode TCL_Transceive(PcbBlock *send, PcbBlock *back);
 	StatusCode TCL_Transceive(TagInfo * tag, byte *sendData, byte sendLen, byte *backData = NULL, byte *backLen = NULL);
-	StatusCode TCL_TransceiveRBlock(TagInfo *tag, bool ack, byte *backData = NULL, byte *backLen = NULL);
+	StatusCode TCL_TransceiveRBlock(TagInfo *tag, bool ack, byte *backData = NULL, byte *backLen = NULL, byte* linked = NULL);
```
Comment (Rotzbua) [inline src/MFRC522Extended.h]: please use c++11 `nullptr` instead of `NULL`

File: src/MFRC522Extended.cpp [hunk 3]
```diff
@@ -807,17 +806,14 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
 		out.inf.size = 0;
 		out.inf.data = NULL;
 	}
-
 	// Initialize the receiving data
 	// TODO Warning: Value escapes the local scope
 	in.inf.data = outBuffer;
 	in.inf.size = outBufferSize;
-
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: please keep the space

Comment (Sleepychord) [inline src/MFRC522Extended.cpp]: I am very sorry to make such a silly mistake and I think I have fixed it.

File: src/MFRC522Extended.cpp (hunk 1 above)
Comment (Sleepychord) [inline src/MFRC522Extended.cpp]: I am very sorry to make such a silly mistake and I think I have fixed it.

File: src/MFRC522Extended.cpp [hunk 4]
```diff
@@ -828,19 +824,19 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
...
+	
 	// Check chaining
 	if ((in.prologue.pcb & 0x10) == 0x00)
 		return result;
-
 	// Result is chained
 	// Send an ACK to receive more data
 	// TODO: Should be checked I've never needed to send an ACK
+
 	while (in.prologue.pcb & 0x10) {
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: In `TCL_TransceiveRBlock` the var `linked` is set:
```c++
*linked = in.prologue.pcb & 0x10;
```
and then at the end of the while it is checked
```c++
if(!linked) break; 
```
But that's the same as
```c++
while (in.prologue.pcb & 0x10)
```

Or did I missed something?

Comment (Sleepychord) [inline src/MFRC522Extended.cpp]: actually, the variables "in"s are local variables. So `while (in.prologue.pcb & 0x10)` in function `TCL_Transceive` will always be true although `in.prologue.pcb` in function `TCL_TransceiveRBlock ` will change.

**Comment by Rotzbua** [comment]
In 2-3 weeks I will have a closer look on it. Thanks so far for support :)

File: src/MFRC522Extended.cpp [hunk 5]
```diff
@@ -828,19 +824,19 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
...
 		byte ackData[FIFO_SIZE];
 		byte ackDataSize = FIFO_SIZE;
-
-		result = TCL_TransceiveRBlock(tag, true, ackData, &ackDataSize);
+		byte linked = 0;
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: `linked` seems more a boolean and should be

File: src/MFRC522Extended.cpp (hunk 4 above)
Comment (Rotzbua) [reply src/MFRC522Extended.cpp]: Ah, I see. You are right.

File: src/MFRC522Extended.cpp [hunk 6]
```diff
@@ -859,7 +856,7 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
 /**
  * Send R-Block to the PICC.
  */
-MFRC522::StatusCode MFRC522Extended::TCL_TransceiveRBlock(TagInfo *tag, bool ack, byte *backData, byte *backLen)
+MFRC522::StatusCode MFRC522Extended::TCL_TransceiveRBlock(TagInfo *tag, bool ack, byte *backData, byte *backLen, byte* linked)
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: `bool* linked`

File: src/MFRC522Extended.cpp [hunk 7]
```diff
@@ -913,7 +911,8 @@ MFRC522::StatusCode MFRC522Extended::TCL_TransceiveRBlock(TagInfo *tag, bool ack
 		*backLen = in.inf.size;
 		memcpy(backData, in.inf.data, in.inf.size);
 	}
-	
+	if(linked)
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: I suggest for bool:
```c++
if(linked) {
     *linked = ((in.prologue.pcb & 0x10)!=0x00);
}

File: src/MFRC522Extended.h [hunk 8]
```diff
@@ -93,7 +93,7 @@ class MFRC522Extended : public MFRC522 {
 	/////////////////////////////////////////////////////////////////////////////////////
 	StatusCode TCL_Transceive(PcbBlock *send, PcbBlock *back);
 	StatusCode TCL_Transceive(TagInfo * tag, byte *sendData, byte sendLen, byte *backData = NULL, byte *backLen = NULL);
-	StatusCode TCL_TransceiveRBlock(TagInfo *tag, bool ack, byte *backData = NULL, byte *backLen = NULL);
+	StatusCode TCL_TransceiveRBlock(TagInfo *tag, bool ack, byte *backData = NULL, byte *backLen = NULL, byte* linked = nullptr);
```
Comment (Rotzbua) [inline src/MFRC522Extended.h]: `bool* linked`

File: src/MFRC522Extended.cpp [hunk 9]
```diff
@@ -701,13 +702,11 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(PcbBlock *send, PcbBlock *ba
 
 		outBufferOffset += 2;
 	}
-
 	// Transceive the block
 	result = PCD_TransceiveData(outBuffer, outBufferOffset, inBuffer, &inBufferSize);
 	if (result != STATUS_OK) {
 		return result;
 	}
-
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: please keep the space

File: src/MFRC522Extended.cpp [hunk 10]
```diff
@@ -701,13 +702,11 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(PcbBlock *send, PcbBlock *ba
 
 		outBufferOffset += 2;
 	}
-
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: please keep the space

File: src/MFRC522Extended.cpp [hunk 11]
```diff
@@ -807,17 +806,14 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
 		out.inf.size = 0;
 		out.inf.data = NULL;
 	}
-
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: please keep the space

File: src/MFRC522Extended.cpp [hunk 12]
```diff
@@ -828,19 +824,19 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
 		*backLen = in.inf.size;
 		memcpy(backData, in.inf.data, in.inf.size);
 	}
-
+	
 	// Check chaining
 	if ((in.prologue.pcb & 0x10) == 0x00)
 		return result;
-
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: please keep the space

File: src/MFRC522Extended.cpp [hunk 13]
```diff
@@ -828,19 +824,19 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
 		*backLen = in.inf.size;
 		memcpy(backData, in.inf.data, in.inf.size);
 	}
-
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: please keep the space

File: src/MFRC522Extended.cpp [hunk 14]
```diff
@@ -807,17 +806,14 @@ MFRC522::StatusCode MFRC522Extended::TCL_Transceive(TagInfo *tag, byte *sendData
...
 	// Initialize the receiving data
 	// TODO Warning: Value escapes the local scope
 	in.inf.data = outBuffer;
 	in.inf.size = outBufferSize;
-
 	result = TCL_Transceive(&out, &in);
 	if (result != STATUS_OK) {
 		return result;
 	}
-
```
Comment (Rotzbua) [inline src/MFRC522Extended.cpp]: please keep the space

**Pull request closed without merging**